### DISCUSSION TITLE
Release Process

### DISCUSSION THREAD

**Discussion by beadon** [David00/rpi-power-monitor#72] [General]
Recommendation:

With each release could we attach the binary to the release?

This will prevent clutter in the branches
This will also allow a merge to a temporary release branch, for testing to occur, before a final merge down to the master branch.

Then the release branch can be deleted, the specific feature branch can then also be deleted.
A develop branch can be maintained in parallel.

**Comment by David00** [comment]
Thank you for starting this - when I made the v0.2.0 release, GitHub was blocking the OS image due to its file size, which led me down a rabbit hole of [Git LFS](https://docs.github.com/en/repositories/working-with-files/managing-large-files/about-large-files-on-github)..

I was able to add the OS image in the v0.1.0 release assets, so I just assumed GitHub had changed the limitations since that upload. However, I re-read the LFS docs and noticed the distinction that LFS is only used for tracking large files, and also found the section about [distributing large binaries](https://docs.github.com/en/repositories/working-with-files/managing-large-files/about-large-files-on-github#distributing-large-binaries).

Turns out there's a separate upload box to use for binaries when creating a release. Somehow I overlooked it when pushing v0.2.0 out.  So, I just added the OS image and the MD5 hash to the v0.2.0 release assets. Thanks for bringing this up - I might have continued with the assumption that I needed LFS if you hadn't!

**Comment by beadon** [comment]
Great!  And no longer a need for gitlfs, right?
